auth/auth.go

```diff
@@ -31,29 +31,32 @@ type Resource struct {
 	Name string
 	// Type of resource, e.g.
 	Type string
+	// Endpoint resource e.g NotesService.Create
+	Endpoint string
 }
 
-// Role an account has
-type Role struct {
-	// Name of the role
-	Name string
-	// The resource it has access
-	// TODO: potentially remove
-	Resource *Resource
+// Token can be short or long lived
+type Token struct {
+	// The token itself
+	Token string `json:"token"`
+	// Type of token, e.g. JWT
+	Type string `json:"type"`
+	// Time of token creation
+	Created time.Time `json:"created"`
+	// Time of token expiry
+	Expiry time.Time `json:"expiry"`
 }
 
 // Account provided by an auth provider
 type Account struct {
 	// ID of the account (UUIDV4, email or username)
-	Id string `json:"id"`
+	ID string `json:"id"`
 	// Token used to authenticate
-	Token string `json:"token"`
-	// Time of Account creation
-	Created time.Time `json:"created"`
-	// Time of Account expiry
-	Expiry time.Time `json:"expiry"`
+	Token Token `json:"token"`
```

Comment:
Token (aka jwt) would not be stored in the account itself. We'd only have secret. Tokens would be created and its up to the client side to store/make use of those.